src/models/iceflow/SIA2D/SIA2D.jl

```diff
@@ -24,6 +23,7 @@ A mutable struct representing a 2D Shallow Ice Approximation (SIA) model.
 - `dSdx::Union{Matrix{R}, Nothing}`: Surface slope in the x-direction.
 - `dSdy::Union{Matrix{R}, Nothing}`: Surface slope in the y-direction.
 - `D::Union{Matrix{R}, Nothing}`: Diffusivity.
+- `D_is_provided::Union{Bool, Nothing}`: Specifies if D is provided by user or computed.
```

Comment:
OK for now, to be adapted with what we discussed yesterday.

Reply:
Yes, we need to think how to provide this flag.